From 6372547053091402976a595834cbbbd673889c7c Mon Sep 17 00:00:00 2001
From: Gleb Kodinets <gkodinets@habana.ai>
Date: Wed, 28 Jul 2021 14:29:02 +0300
Subject: [PATCH] [SW-23529] greco: pci: lineup arc and arm bar mapping

Making BAR mapping function same in greco arm and arc
to prepare before making code sharing instead of code duplication.

Change-Id: Id79cc32bd922b37728ebaee3e65dfc55e0ee5cf3
---

diff --git a/arch/greco/arm/asic.c b/arch/greco/arm/asic.c
index 3576b67..f9cb4a6 100644
--- a/arch/greco/arm/asic.c
+++ b/arch/greco/arm/asic.c
@@ -89,7 +89,7 @@
 	/* Inbound Region 0 - Bar - - Point to SRAM and CFG */
 	inbound_region.mode = PCI_BAR_MATCH_MODE;
 	inbound_region.bar = PCI_BAR_ID_0;
-	inbound_region.addr = SDRAM_BASE;
+	inbound_region.addr = SDRAM_FULL_BASE;
 	hl_pci_set_inbound_region(0, &inbound_region);
 
 	/* Inbound Region 1 - Bar 4 - Point to DRAM */
diff --git a/arch/greco/arm/common.h b/arch/greco/arm/common.h
index 8f3131f..8ea4d58 100644
--- a/arch/greco/arm/common.h
+++ b/arch/greco/arm/common.h
@@ -32,6 +32,7 @@
 #define HL_SPI_BASE			0x7FF8000000ull
 #define HL_CFG_BASE			0x7FFC000000ull
 #define SDRAM_BASE			0x7FF0000000ull
+#define SDRAM_FULL_BASE			0x7FF0000000ull
 #define DRAM_BASE			0x1000000000000ull
 #define BOOT_ROM_BASE			0x7FFB040000ull
 #define BOOT_ROM_SIZE			0x10000
